chia/util/api_decorators.py

```diff
@@ -22,40 +29,54 @@ def f_substitute(*args, **kwargs):
             if param_name != "return" and isinstance(inter[param_name], Streamable):
                 if param_class.__name__ == "bytes":
                     continue
-                if hasattr(f, "bytes_required"):
+                if hasattr(f, BYTES_REQUIRED):
                     inter[f"{param_name}_bytes"] = bytes(inter[param_name])
                     continue
             if param_name != "return" and isinstance(inter[param_name], bytes):
                 if param_class.__name__ == "bytes":
                     continue
-                if hasattr(f, "bytes_required"):
+                if hasattr(f, BYTES_REQUIRED):
                     inter[f"{param_name}_bytes"] = inter[param_name]
-                inter[param_name] = param_class.from_bytes(inter[param_name])
+                if inter[param_name] == b"":
+                    inter[param_name] = None
+                else:
+                    inter[param_name] = param_class.from_bytes(inter[param_name])
         return f(**inter)
 
-    setattr(f_substitute, "api_function", True)
+    setattr(f_substitute, API_FUNCTION, True)
     return f_substitute
 
 
 def peer_required(func):
     def inner():
-        setattr(func, "peer_required", True)
+        setattr(func, PEER_REQUIRED, True)
         return func
 
     return inner()
 
 
 def bytes_required(func):
     def inner():
-        setattr(func, "bytes_required", True)
+        setattr(func, BYTES_REQUIRED, True)
         return func
 
     return inner()
 
 
 def execute_task(func):
     def inner():
-        setattr(func, "execute_task", True)
+        setattr(func, EXECUTE_TASK, True)
         return func
 
     return inner()
+
+
+def reply_type(type):
```

Comment:
Same comment about reply_type - could use a more specific name.

Not a big deal though - unlikely to collide in our codebase!

Reply:
msg_reply_type ?